connectivity/drivers/ble/FEATURE_BLE/TARGET_STM32WB/STM32Cube_FW/shci.h

```diff
@@ -502,7 +502,7 @@ extern "C" {
    * MaxConnEventLength
    * This parameter determines the maximum duration of a slave connection event. When this duration is reached the slave closes
    * the current connections event (whatever is the CE_length parameter specified by the master in HCI_CREATE_CONNECTION HCI command),
-   * expressed in units of 625/256 �s (~2.44 �s)
+   * expressed in units of 625/256 us (~2.44 us)
```

Comment:
µ is UTF-8. The codepoint is U+03BC, which is encoded in UTF-8 as: `0xCE 0xBC` (https://www.compart.com/en/unicode/U+03BC)

Reply:
Good point. I think we should agree on what characters are allowed in Mbed OS. ASCII feels a bit restrictive but is more compatible. If we allow any UTF-8, then the CircleCI needs to be fixed instead.

This particular character was added a few weeks ago, so our code base was likely ASCII-only before that.

Reply:
What breaks exactly? What tool chokes on this? I didn't see it explained in the commit message.

Reply:
The title says "don't allow invalid UTF-8". Does this mean if encoded correctly (in a valid manner), µ would be OK?

Confusingly, the description says "Introducing non-UTF-8 characters", but then removes a UTF-8 character. I'm not sure what is breaking what. Maybe we need to correctly encode `shci.h` as UTF-8 rather than remove the µ?

Reply:
The doc_builder tool breaks when it tries to do a `pathlib.Path.read_text`. If the file it's reading from is saved with utf8 encoding, then µ should work ok.

```
Python 3.9.5 (default, May  4 2021, 03:33:11)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import pathlib
>>> import sys
>>> sys.getdefaultencoding()
'utf-8'
>>> pathlib.Path("test.txt").read_text()
'μ\n'
```

Reply:
Ah so the real issue in encoding of that file?

Reply:
So, rather than replacing µ, we should convert the file to utf-8. For example, in vim you can do https://stackoverflow.com/a/66572180. This keeps the greek letter.

Reply:
Yup the issue is that the file is not encoded as UTF-8:
```file -I shci.h 
shci.h: text/x-c; charset=unknown-8bit```

Happy to try and convert the file to keep the µ